Purchases/Purchasing/StoreKitAbstractions/SubscriptionPeriod.swift

```diff
@@ -64,6 +64,22 @@ import StoreKit
 
 }
 
+extension SubscriptionPeriod {
+    func pricePerMonth(withTotalPrice price: Decimal) -> Decimal {
+        let periodsPerMonth: Decimal = {
+            switch self.unit {
+            case .day: return 1 / 30
+            case .week: return 1 / 4
+            case .month: return 1
+            case .year: return 12
+            case .unknown: return 1
+            }
+        }() * Decimal(self.value)
+
+        return price / periodsPerMonth
```

Comment:
There's a bit of nuance to this result, as explained by the [StoreKit whisperer himelf](https://stackoverflow.com/a/47445051/5074358). 
We need to handle rounding results appropriately. 
I had started on this on an [overly-complicated branch on Friday](https://github.com/RevenueCat/purchases-ios/compare/main...price_as_duration_methods) but I didn't get time to pick it back up this week, sadly. 
I'm more than happy to go with this solution instead, let's just make sure that we handle rounding correctly

Reply:
Oh nice! I'll incorporate that later :)

Reply:
Sorry that looks like we both did the same thing 😬 looks like my implementation is a bit shorter.
I just added the rounding behavior and added a few tests that would have failed before the fix 👍